src/Service.Tests/SqlTests/RestApiTests/Patch/PatchApiTestBase.cs

```diff
@@ -458,10 +449,259 @@ public virtual async Task PatchOneInsertWithDatabasePolicy()
+                    operationType: EntityActionOperation.UpsertIncremental,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.OK,
+                    clientRoleHeader: "test_role_with_excluded_fields"
                 );
         }
 
+        /// <summary>
+        /// Tests that for a successful PATCH API request, the response returned takes into account the database policy configured for the read action.
+        /// URI Path: PK of existing record.
+        /// Req Body: Valid Parameter with intended update.
+        /// Expects:
+        /// Status: 200 OK
+        /// Response Body: Empty. The read action for the role used in this test has a database policy
+        /// defined which states that title cannot be equal to Test. Since, this test updates the title
+        /// to Test the response must be empty.
+        /// </summary>
+        [TestMethod]
+        public virtual async Task Patch_Update_WithReadDatabasePolicyTest()
+        {
+            string requestBody = @"
+            {
+                ""title"": ""Test""
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "id/8",
+                    queryString: null,
+                    entityNameOrPath: _integrationEntityName,
+                    sqlQuery: GetQuery(nameof(PatchOne_Update_NoReadTest)),
+                    operationType: EntityActionOperation.UpsertIncremental,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.OK,
+                    clientRoleHeader: "test_role_with_policy_excluded_fields"
+                );
+        }
+
+        /// <summary>
+        /// Tests that for a successful PATCH API request, the response returned takes into account the database policy configured for the read action.
+        /// URI Path: PK of existing record.
+        /// Req Body: Valid Parameter with intended update.
+        /// Expects:
+        /// Status: 200 OK
+        /// Response Body: Non-Empty and does not contain the publisher_id field. The read action for the role used in this test has a database policy
+        /// defined which states that title cannot be equal to Test. Since, this test updates the title
+        /// to a different the response must be non-empty. Also, since the role excludes the publisher_id field, the repsonse should not
+        /// contain publisher_id field.
+        /// </summary>
+        [TestMethod]
+        public virtual async Task Patch_Update_WithReadDatabasePolicyUnsatisfiedTest()
+        {
+            string requestBody = @"
+            {
+                ""title"": ""Heart of Darkness""
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "id/8",
+                    queryString: null,
+                    entityNameOrPath: _integrationEntityName,
+                    sqlQuery: GetQuery(nameof(Patch_Update_WithExcludeFieldsTest)),
```

Comment:
Why is the query `Patch_Update_WithExcludeFieldsTest`?

Reply:
`Patch_Update_WithExcludeFieldsTest`: This test uses a role that has only `exclude` field and no database policy in the read action. So, in this case, the second roundtrip to database is not performed and the additional fields are removed by DAB. This test is for validating this code path.


`Patch_Update_WithReadDatabasePolicyUnsatisfiedTest`: This test uses a role that has `exclude` field and also a database policy defined for read action. In this case, the database select query takes care of both - honoring the database policy as well as include/exclude fields. This test validates the case when the database policy condition evaluates to `false`.
 So, a non-empty response is expected. Since, both these roles (ones used in  `Patch_Update_WithReadDatabasePolicyUnsatisfiedTest` and `Patch_Update_WithExcludeFieldsTest`) ignore the same field and have the same request body, the expected results are the same. 

So, I've re-used the same database query for both the tests.